components/details/MetadataTags.tsx

```diff
@@ -0,0 +1,67 @@
+import { memo } from "react";
+import Head from "next/head";
+import { useRouter } from "next/router";
+
+export interface MetadataShareTagsProps {
+  title: string;
+  description: string;
+  shareTitle?: string;
+  shareDescription?: string;
+  shareImage?: string;
+  type?: string;
+}
+
+const MetadataTags = ({
+  title,
+  description,
+  shareTitle,
+  shareDescription,
+  shareImage,
+  type,
+}: MetadataShareTagsProps) => {
+  if (shareTitle === undefined) {
+    shareTitle = title;
+  }
+
+  if (shareDescription === undefined) {
+    shareDescription = description;
+  }
+
+  if (shareImage === undefined) {
+    shareImage = `${process.env.NEXT_PUBLIC_HOST}/logo-sharecard.jpg`;
+  }
+
+  if (type === undefined) {
+    type = "website";
+  }
+
+  const pageUrl = `${process.env.NEXT_PUBLIC_HOST}${useRouter().asPath}`;
```

Comment:
This was already a problem, however, if you accidentally set NEXT_PUBLIC_HOST to http://localhost:3000/ rather than http://localhost:3000 it will cause problems. I wonder if we should make a function or use an existing one that can handle joining these parts of the URL in a more robust way? E.g. new URL() maybe? But I haven't tried it in depth.